src/V3Timing.cpp

```diff
@@ -85,18 +90,144 @@ class TimingVisitor final : public VNVisitor {
+    }
+    void visit(AstNodeProcedure* nodep) override {
+        VL_RESTORER(m_procp);
+        m_procp = nodep;
+        iterateChildren(nodep);
+    }
+    void visit(AstCFunc* nodep) override {
+        VL_RESTORER(m_procp);
+        m_procp = nodep;
+        iterateChildren(nodep);
+        DependencyVertex* const vxp = getDependencyVertex(nodep);
+        if (m_classp) {  // If class method (possibly overrides another method)
+            if (!m_classp->user1SetOnce()) m_classp->repairCache();
+            // Go over overridden functions
+            for (auto* cextp = m_classp->extendsp(); cextp;
+                 cextp = VN_AS(cextp->nextp(), ClassExtends)) {
+                // TODO: It is possible that a methods the same name in the base class is not
+                // actually overridden by our method. If this causes a problem, traverse to the
+                // root of the inheritance hierarchy and check if the original method is virtual or
+                // not.
+                if (!cextp->classp()->user1SetOnce()) cextp->classp()->repairCache();
+                if (auto* const overriddenp
+                    = VN_CAST(cextp->classp()->findMember(nodep->name()), CFunc)) {
+                    if (overriddenp->user2()) {  // If it's suspendable
+                        nodep->user2(true);  // Then we are also suspendable
+                        // As it's suspendable already, there is no need to add it as our
+                        // dependency or self to its dependencies
+                    } else {
+                        // Make a dependency cycle, as being suspendable should propagate both up
+                        // and down the inheritance tree
+                        DependencyVertex* const overriddenVxp = getDependencyVertex(overriddenp);
+                        makeDependencyEdge(vxp, overriddenVxp);
+                        makeDependencyEdge(overriddenVxp, vxp);
+                    }
+                }
+            }
+        }
+        // If suspendable, propagate that to its dependents
+        if (nodep->user2()) propagateSuspendable(vxp);
+    }
+    void visit(AstNodeCCall* nodep) override {
+        if (nodep->funcp()->user2()) {
+            m_procp->user2(true);
+        } else {
+            DependencyVertex* const procVxp = getDependencyVertex(m_procp);
+            DependencyVertex* const funcVxp = getDependencyVertex(nodep->funcp());
+            makeDependencyEdge(procVxp, funcVxp);
+            iterateChildren(nodep);
+        }
+    }
+    void visit(AstBegin* nodep) override {
+        VL_RESTORER(m_procp);
+        m_procp = nodep;
+        iterateChildren(nodep);
+    }
+    void visit(AstNode* nodep) override {
+        if (nodep->isTimingControl()) {
+            v3Global.setUsesTiming();
+            if (m_procp) m_procp->user2(true);
+        }
+        iterateChildren(nodep);
+    }
+
+    //--------------------
+    void visit(AstNodeExpr*) override {}  // Accelerate
```

Comment:
Might be possible to have a timing control embedded in a statement inside an expression.

Reply:
Is it possible? All timing controls at this point are either `AstNodeStmt` or `AstNodeBlock`. `AstCAwait` is an expression, but it only exists after `TimingVisitor`.